worker/crlWorker/crlWorker.go

```diff
@@ -0,0 +1,117 @@
+package crlWorker
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/mozilla/tls-observatory/certificate"
+	"github.com/mozilla/tls-observatory/worker"
+)
+
+var (
+	workerName = "crlWorker"
+	workerDesc = "Checks certificate CRL (Certificate Revocation List) and reports on certificate revocation"
+)
+
+type Result struct {
+	Revoked bool `json:"revoked"`
+}
+
+type eval struct{}
+
+// Run implements the worker.Worker interface and is called check CRL status
+func (e eval) Run(in worker.Input, resChan chan worker.Result) {
+	result := worker.Result{
+		WorkerName: workerName,
+	}
+	crlRes := Result{}
+
+	// Grab first CRL response we can from our certificate
+	crlResp, err := e.getFirstCRLData(in.Certificate)
+	if err != nil {
+		result.Errors = append(result.Errors, err.Error())
+	}
+	certList, err := x509.ParseCRL(crlResp)
+	if err != nil {
+		result.Errors = append(result.Errors, err.Error())
+	}
+
+	// Verify our CRL response is signed properly
+	x509Cert, err := in.Certificate.ToX509()
+	if err != nil {
+		result.Errors = append(result.Errors, err.Error())
+	}
+	if err := x509Cert.CheckCRLSignature(certList); err != nil {
+		result.Errors = append(result.Errors, err.Error())
+	}
+
+	// Check if our certificate is in the revoked list
+	revoked := certList.TBSCertList.RevokedCertificates
+	for i := range revoked {
+		if x509Cert.SerialNumber.Cmp(revoked[i].SerialNumber) == 0 {
+			// certificate is in revoked list, serials match
+			crlRes.Revoked = true
+			break
+		}
+	}
+
+	// Marshal the response
+	bs, err := json.Marshal(&crlRes)
+	if err != nil {
+		result.Success = false
+		result.Errors = append(result.Errors, err.Error())
+	} else {
+		result.Success = true
+		result.Result = bs
+	}
+	resChan <- result
```

Comment:
I think it'd be useful to add a column to the certificate table to store the fact it has been revoked via CRL.